File: Know Basic maths/python/count_digits.py
def count_dig(num: int) -> int:
    if num == 0: return 1
    c = 0
    while num > 0:
        num = num // 10
        c += 1
    return c

n = int(input("Enter a number: "))
print("The number of digits are: " , count_dig(n))

# more efficient and better way 

import math

def count_digits_log(num: int) -> int:
    if num == 0:
        return 1
    return math.floor(math.log10(num)) + 1

print("The number of digits are:", count_digits_log(12345))

File: Know Basic maths/C++/count_digits.cpp

#include <iostream>
using namespace std;


int countDigits(int n){
    int cnt = 0;
    while(n > 0){
        cnt = cnt + 1;
        n = n / 10;
    }
    return cnt;
}



int main() {
    int N;
    cout << "Enter the number N: ";
    cin >> N;
    int digits = countDigits(N);
    cout << "Number of Digits in N: "<< digits << endl;
    return 0;
}

                                